rclcpp/src/rclcpp/node_interfaces/node_topics.cpp

```diff
@@ -14,9 +14,18 @@
 
 #include "rclcpp/node_interfaces/node_topics.hpp"
 
+#include <stdexcept>
```

Comment:
throughout the PR there are a lot of changes to unrelated files where you fix some header include issues.
This PR is already very complex, so I would suggest to undo these changes and open a separate PR for them

Reply:
That's my fault, I noticed them while we were developing them and just fixed them.  I'm fine with pulling them out into a separate PR.

Reply:
I've now removed the unrelated changes, and I'll submit them separately from this PR.